### ISSUE TITLE
IBRUN fails on Frontera

### ISSUE THREAD

**Issue by AymenFJA** [radical-cybertools/radical.pilot#2583]
```shell
  python               : /scratch1/xxxx/rpilot/aymen/ve/raptor/bin/python3
  pythonpath           : /opt/apps/intel19/impi19_0/python3/3.7.0/lib/python3.7/site-packages
  version              : 3.7.0
  virtualenv           : /scratch1/xxxx/rpilot/aymen/ve/raptor

  radical.gtod         : 1.13.0
  radical.pilot        : 1.14.0-v1.14.0-5-g7b033bf49@feature-parsl_raptor_executor (based on devel)
  radical.saga         : 1.14.0
  radical.utils        : 1.14.0
```

Using RAPTOR on Frontera with the current `resource_tacc.json` : 
```json
{
    "frontera_interactive": {
        "description"                 : "Petascale computing system at the Texas Advanced Computing Center (TACC)",
        "notes"                       : "Specify this resource in the pilot description and run RP within a compute node(s)",
        "schemas"                     : ["local", "ssh"],
        "ssh"                         :
        {
            "job_manager_endpoint"    : "ssh://localhost/",
            "filesystem_endpoint"     : "sftp://localhost/"
        },
        "local"                       :
        {
            "job_manager_endpoint"    : "fork://localhost/",
            "filesystem_endpoint"     : "file://localhost/"
        },
        "resource_manager"            : "SLURM",
        "cores_per_node"              : 56,
        "gpus_per_node"               : 0,
        "agent_config"                : "default",
        "agent_scheduler"             : "CONTINUOUS",
        "agent_spawner"               : "POPEN",
        "launch_methods"              : {
                                         "order": ["IBRUN"],
                                         "IBRUN" : {
                                             "pre_exec_cached": ["module load TACC",
                                                                 "module unload intel",
                                                                 "module unload impi",
                                                                 "module load intel",
                                                                 "module load impi"]
                                         }
                                        },
        "pre_bootstrap_1"             :["export RP_APP_TUNNEL_ADDR=144.76.72.175:27017",
                                        "echo  $RP_APP_TUNNEL_ADDR"],
        "pre_bootstrap_0"             :["module unload intel",
                                        "module unload impi",
                                        "module load   intel",
                                        "module load   impi",
                                        "module load   python3/3.9.2"],

        "default_remote_workdir"      : "$SCRATCH",
        "valid_roots"                 : ["$SCRATCH", "$WORK"],
        "rp_version"                  : "local",
        "virtenv_mode"                : "create",
        "python_dist"                 : "default",
        "task_pre_exec"               : []
    },
```
leads to 2 errors:

1- `AttributeError: 'IBRun' object has no attribute` which is fixed by @mtitov in this PR https://github.com/radical-cybertools/radical.pilot/pull/2580/commits/f8a3ce2eca0ebdef5e36b1a4407b7bb8ce8a48dd

2- `ERROR: Number of hosts (requested through -n) plus offset (via -o) exceeds the total number of slots available.`

This is the command that RP is using to launch the master : `/usr/local/bin/ibrun -n 1 -o 0 $RP_TASK_SANDBOX/master.000000.exec.sh`

**Comment by mtitov** [comment]
@AymenFJA is this interactive mode? and do you set option `-n`? That option shouldn't be set or be set with max number of tasks (`-n 56` or `--ntasks-per-node=56`)

**Comment by AymenFJA** [comment]
@mtitov I do not use `idev` instead i use this command `srun --pty -A XXXXX -N 2 -n 56 -t 01:00:00 -p development /bin/bash -l #`. Also I use `SBATCH` script with the following arguments: 

```bash
#!/bin/bash
#----------------------------------------------------

#SBATCH -J colmena           # Job name
#SBATCH -o colmena.o%j       # Name of stdout output file
#SBATCH -e colmena.e%j       # Name of stderr error file
#SBATCH -p development       # Queue (partition) name
#SBATCH -N 2                 # Total # of nodes
#SBATCH -n 56                # Total # of mpi tasks
#SBATCH -t 02:00:00          # Run time (hh:mm:ss)
#SBATCH -A XXXXXX            # Project/Allocation name (req'd if you have more than 1)
```

**Comment by mtitov** [comment]
@AymenFJA  couldn't reproduce with interactive mode when options `-n 1 -o 0` for IBRun are set. Can you please try to use `--ntasks-per-node` instead of `-n` and for master task description define `"pre_launch": "export TACC_IBRUN_DEBUG=1"`, and attach a resulted pilot session. Thanks!

**Comment by AymenFJA** [comment]
@mtitov the session folder zipper was way too big (160 MB) so I attached the `master.00000` only here.
[master.000000.zip](https://github.com/radical-cybertools/radical.pilot/files/8564708/master.000000.zip)

let me know if this is not enough.

**Comment by AymenFJA** [comment]
This is fixed with #2597
